core/shared/src/main/scala/zio/stm/ZSTM.scala

```diff
@@ -845,10 +845,7 @@ object ZSTM {
    * transactional effect that produces a list of values.
    */
   def collectAll[E, A](i: Iterable[STM[E, A]]): STM[E, List[A]] =
```

Comment:
I believe there was a bug here where effects were evaluated in the wrong order. I added a fix for it and a test.